app/views/directives/create-secret.html

```diff
@@ -350,6 +350,9 @@
         </div>
       </div>
     </div>
+    <div ng-if="newSecret.type === 'generic'">
+      <edit-config-map-or-secret model="newSecret.data.genericKeyValues" type="secret"></edit-config-map-or-secret>
```

Comment:
Does this work if you use...?

```
model="newSecret.data"
```

Then you don't need the intermediate `genericKeyValues` property.

Reply:
Unfortunately, it does not work as the `editConfigMapOrSecret` directive expects a `model.data` to exist, and it replaces all the values in `data` to match the key/value pairs entered.  This effectively replaces `newSecret.data.secretName` (and any other values we may use in data).